sbt-plugin/src/main/scala/ch/epfl/scala/AnalyzeDependencyGraph.scala

```diff
@@ -0,0 +1,220 @@
+    } else None
+  }
+
+  private def getGithubTokenFromGhConfigDir(): String = {
+    val ghConfigDir =
+      Properties.envOrElse("GH_CONFIG_DIR", Paths.get(System.getProperty("user.home"), ".config", "gh").toString)
+    val ghConfigFile = Paths.get(ghConfigDir).resolve("hosts.yml").toFile
+    getGithubToken(ghConfigFile).getOrElse {
+      val ghConfigPath =
+        Properties.envOrElse("HUB_CONFIG", Paths.get(System.getProperty("user.home"), ".config", "hub").toString)
+      val hubConfigFile = Paths.get(ghConfigPath).toFile
+      getGithubToken(hubConfigFile).getOrElse(githubToken())
+    }
+  }
+
+  private def downloadAlerts(state: State, repo: String): Try[State] = {
+    val snapshotUrl = s"https://api.github.com/repos/$repo/dependabot/alerts"
+    val request =
+      Gigahorse.url(snapshotUrl).get.addHeaders("Authorization" -> s"token ${getGithubTokenFromGhConfigDir()}")
+    state.log.info(s"Downloading alerts from $snapshotUrl")
+    for {
+      httpResp <- Try(Await.result(http.processFull(request), Duration.Inf))
+      vulnerabilities <- getVulnerabilities(httpResp)
+    } yield {
+      state.log.info(s"Downloaded ${vulnerabilities.size} alerts")
+      state.put(githubAlertsKey, vulnerabilities)
+    }
+  }
+
+  private def getAllArtifacts(state: State): Seq[String] =
+    getGithubManifest(state).flatMap { manifests =>
+      manifests.flatMap {
+        case (_, manifest) =>
+          manifest.resolved.values.toSeq.map(_.package_url)
+      }
+    }.distinct
+
+  private def translateToSemVer(string: String): String =
+    string.replaceAll("([a-zA-Z]+)", "0").replaceAll("([0-9]+)\\.([0-9]+)\\.([0-9]+)\\.([0-9]+)", "$1.$2.$3-$4")
+
+  private def versionMatchesRange(versionStr: String, rangeStr: String): Boolean = {
+    val range = rangeStr.replaceAll(" ", "").replace(",", " ")
+    VersionNumber(translateToSemVer(versionStr)).matchesSemVer(SemanticSelector(translateToSemVer(range)))
+  }
+
+  private def vulnerabilityMatchesArtifact(alert: Vulnerability, artifact: String): Vulnerable = {
+    val alertMavenPath = s"pkg:maven/${alert.packageId.replace(":", "/")}@"
+    if (artifact.startsWith(alertMavenPath)) {
+      val version = artifact.replaceAll(".*@", "")
+      if (versionMatchesRange(version, alert.vulnerableVersionRange)) Bad else Good
+    } else No
+  }
+
+  private def vulnerabilityMatchesArtifacts(
+      alert: Vulnerability,
+      artifacts: Seq[String]
+  ): Map[Vulnerable, Seq[String]] =
+    artifacts.foldLeft(Map(Good -> Seq.empty[String], Bad -> Seq.empty[String])) { (acc, artifact) =>
+      val res = vulnerabilityMatchesArtifact(alert, artifact)
+      if (res != No) acc.updated(res, acc(res) :+ artifact) else acc
+    }
+
+  private def analyzeCves(state: State): State = {
+    val vulnerabilities =
+      getStateOrWarn(state, githubAlertsKey, "artifcats", s"${AnalyzeDependencies} alerts").getOrElse(Seq.empty)
```

Comment:
This should take the vulnerabilities as arguments:

```suggestion
  private def analyzeCves(state: State, vulnerabilities: Seq[Vulnerability]): State = {
    val vulnerabilities =
      getStateOrWarn(state, githubAlertsKey, "artifcats", s"${AnalyzeDependencies} alerts").getOrElse(Seq.empty)
```